Plugins/Player/Player.cpp

```diff
@@ -1336,4 +1344,171 @@ ArgumentStack Player::SetResManOverride(ArgumentStack&& args)
     return Services::Events::Arguments();
 }
 
+
+CExoLocString CreateCExoLocString(const std::string& str)
+{
+    CExoLocString locStr;
+    locStr.AddString(0,CExoString(str.c_str()),0);
+    return locStr;
+}
+
+ArgumentStack Player::AddCustomJournalEntry(ArgumentStack&& args)
+{
+    int32_t retval = -1;
+    if (auto *pPlayer = player(args))
+    {
+        auto *pCreature = Globals::AppManager()->m_pServerExoApp->GetCreatureByGameObjectID(pPlayer->m_oidNWSObject);
+        
+        if (pCreature && pCreature->m_pJournal)
+        {
+            auto entries = pCreature->m_pJournal->m_lstEntries;
+            
+            const auto questName = Services::Events::ExtractArgument<std::string>(args);
+            const auto questText = Services::Events::ExtractArgument<std::string>(args);
+            const auto tag = Services::Events::ExtractArgument<std::string>(args);
+            const auto state = Services::Events::ExtractArgument<int32_t>(args);
+            const auto priority = Services::Events::ExtractArgument<int32_t>(args);
+            const auto completed = Services::Events::ExtractArgument<int32_t>(args);
+            const auto diplayed = Services::Events::ExtractArgument<int32_t>(args);
+            const auto updated = Services::Events::ExtractArgument<int32_t>(args);
```

Comment:
Some `ASSERT_OR_THROW()`s would be nice here. Especially for the `SJournalEntry` stuff that takes an unsigned value.

Also have you tested what happens when `questName`, `questText` or `tag` are empty strings?

Reply:
Added ASSERT OR THROW on the tag - my tests showed that the tag is the only parameter that is actually required for success.

Reply:
```c
    uint32_t nCalendarDay;
    uint32_t nTimeOfDay;
    uint32_t nState;
    uint32_t nPriority;
    uint16_t nPictureIndex;
```

These are all unsigned, should have an `ASSERT_OR_THROW(var >= 0)`

Reply:
added extra combined assert or throw-  now they are all checked to be >= 0

Reply:
Haha, sorry D: but we prefer them not combined so it's easier to see which assert is being thrown.

Reply:
lol - done
Reason I kept the 'tag' assertion higher up - is because it is the bare minimum required for the method to function

Reply:
actually done now: not sure what happened last night - must have forgot to push my changes